### PR TITLE
Allocator crate

### PR DESCRIPTION
**Pull request by b00f**
## Description

Adding new crate for allocating memory and updating the example.

## Checklist

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] CHANGELOG is updated.

### CODE REVIEW

File: crates/kelk/src/storage/collections/bst/tree.rs [hunk 1]
```diff
@@ -141,7 +141,7 @@ where
                 }
                 offset = node.left;
             } else {
-                if node.left.eq(&0) {
+                if node.right.eq(&0) {
```
Comment (themantre) [inline crates/kelk/src/storage/collections/bst/tree.rs:144]: write test for this condition.

File: crates/allocator/src/lib.rs [hunk 2]
```diff
@@ -22,15 +20,15 @@ pub fn panic(_info: &::core::panic::PanicInfo) -> ! {
...
 /// the execution with trap.
-#[cfg(not(test))]
+#[cfg(no_std)]
 #[alloc_error_handler]
 #[no_mangle]
 fn oom(_: core::alloc::Layout) -> ! {
     ::core::intrinsics::abort();
 }
 
 /// Needed for non-wasm targets.
```
Comment (themantre) [inline crates/allocator/src/lib.rs:30]: Update the comment

File: examples/token/ReadMe.md [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
+```
 cargo +nightly build --target wasm32-unknown-unknown --release -Z unstable-options --out-dir ./wasm
 ```
 
+It is recommended to remove absolute paths from the WASM binary. Check this issue for more information: https://github.com/rust-lang/rust/issues/40552
+```
+RUSTFLAGS="--remap-path-prefix=$(realpath ../../)=kelk --remap-path-prefix=$HOME/.cargo=cargo --remap-path-prefix=$HOME/.rustup=rustup" cargo +nightly build --target wasm32-unknown-unknown --release -Z unstable-options --out-dir ./wasm
+```
+
+You can check if the absolute path is removed from the binary:
```
Comment (themantre) [inline examples/token/ReadMe.md]: ... paths are ...

**Review by themantre — CHANGES_REQUESTED** [review]

File: crates/allocator/src/lib.rs (hunk 2 above)
Comment (b00f) [reply crates/allocator/src/lib.rs:30]: Comments are from here: https://github.com/rustwasm/wee_alloc/blob/master/example/src/lib.rs

**Pull request merged**